src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1311,11 +1302,11 @@ public static int LastIndexOfAny(ref byte searchSpace, byte value0, byte value1,
 
         // Optimized byte-based SequenceEquals. The "length" parameter for this one is declared a nuint rather than int as we also use it for types other than byte
         // where the length can exceed 2Gb once scaled by sizeof(T).
-        public static bool SequenceEqual(ref byte first, ref byte second, nuint length)
+        public static unsafe bool SequenceEqual(ref byte first, ref byte second, nuint length)
         {
             bool result;
             // Use nint for arithmetic to avoid unnecessary 64->32->64 truncations
-            if (length >= sizeof(nuint))
+            if (length >= (nuint)sizeof(nuint))
```

Comment:
This seems like something the compiler should allow as well...

It allowed `ulong >= sizeof()` but isn't allowing `nuint >= sizeof()`, given that `sizeof()` is guaranteed to be positive, this seems like it should "just work".

Reply:
`sizeof(ulong)` is a constant whereas `sizeof(nuint)` is an `int` calculated at runtime, and `>=` is ambiguous when the arguments are `nuint` and `int` (or `ulong` and `int`).

---
In reply to: [422452236](https://github.com/dotnet/runtime/pull/36159#discussion_r422452236) [](ancestors = 422452236)

Reply:
> sizeof(nuint) is an int calculated at runtime

is that the language specification? The runtime spec explicitly states that `sizeof` is a `uint`...
![image](https://user-images.githubusercontent.com/10487869/81464628-1acd8100-9178-11ea-94b4-e64a890b28a1.png)

Reply:
Ah, I see, it is here: https://github.com/dotnet/csharplang/blob/a5e393b018b04dfa55aae0000290ca087b508495/spec/unsafe-code.md#the-sizeof-operator
> The result of the sizeof operator is a value of type int. For certain predefined types, the sizeof operator yields a constant value as shown in the table below.